src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -99,10 +102,10 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
 
                 String highlighterType = field.fieldOptions().highlighterType();
                 if (highlighterType == null) {
-                    boolean useFastVectorHighlighter = fieldMapper.fieldType().storeTermVectors() && fieldMapper.fieldType().storeTermVectorOffsets() && fieldMapper.fieldType().storeTermVectorPositions();
+                    boolean useFastVectorHighlighter = highlighters.get("fvh").canHighlight(fieldMapper);
                     if (useFastVectorHighlighter) {
                         highlighterType = "fvh";
-                    } else if (fieldMapper.fieldType().indexOptions() == IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS) {
+                    } else if (highlighters.get("postings").canHighlight(fieldMapper)) {
```

Comment:
maybe overkill, but is it worth keeping a list of highlighters ordered by precedence instead? it could even allow plugins to provide their own precedence, but that's maybe too much? I am not sure, just asking